src/Microsoft.Identity.Client/Event.cs

```diff
@@ -58,5 +61,31 @@ public void Stop()
         {
             this[ElapsedTime] = (CurrentUnixTimeMilliseconds() - _startTimestamp).ToString();  // It is a duration
         }
+
+        public static string ScrubTenant(Uri uri) // Note: There is also a Unit Test case for this helper
+        {
+            if (!uri.IsAbsoluteUri)
+            {
+                throw new ArgumentException("Requires an absolute uri");
+            }
+            // only collect telemetry for well-known hosts, // omit B2C ???
+            if (!AadAuthority.IsInTrustedHostList(uri.Host)) // || uri.AbsolutePath.StartsWith("/" + B2CAuthority.Prefix))
```

Comment:
This condition is currently different than [what Android Telemetry has implemented](https://github.com/AzureAD/microsoft-authentication-library-for-android/blob/4808844266d8790adf5b988d9ea35c1d83e30ce7/msal/src/telemetry/java/com/microsoft/identity/client/HttpEvent.java#L53-L56). Looks like we do not have to completely omit the B2C case, given that this helper is capable to mask the tenant from B2C authority url. Read[this test case](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/326/files#diff-529317749eb00d3b13a94449be1fdddeR170) to see what it can do.